tool/tctl/common/plugin/awsic.go

```diff
@@ -316,3 +317,58 @@ func (p *PluginsCommand) InstallAWSIC(ctx context.Context, args installPluginArg
 
 	return nil
 }
+
+type awsICUpdateCredsArgs struct {
+	cmd        *kingpin.CmdClause
+	pluginName string
+	payload    string
+}
+
+func (p *PluginsCommand) initUpdateCredsAWSIC(parent *kingpin.CmdClause) {
+	p.updateCreds.awsic.cmd = parent.Command("awsic", "Update the AWS Identity Center SCIM bearer token")
+	cmd := p.updateCreds.awsic.cmd
+
+	cmd.Flag("plugin-name", "Name of the AWSIC plugin instance to update. Defaults to "+apicommon.OriginAWSIdentityCenter+".").
+		Default(apicommon.OriginAWSIdentityCenter).
+		StringVar(&p.updateCreds.awsic.pluginName)
+
+	cmd.Arg("token", "The new SCIM bearer token.").
+		PlaceHolder("TOKEN").
+		Required().
+		StringVar(&p.updateCreds.awsic.payload)
+}
+
+func (p *PluginsCommand) UpdateAWSICCreds(ctx context.Context, args installPluginArgs) error {
+	plugin, err := args.plugins.GetPlugin(ctx, &pluginsv1.GetPluginRequest{
+		Name:        p.updateCreds.awsic.pluginName,
+		WithSecrets: true,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	staticCredsRef := plugin.Credentials.GetStaticCredentialsRef()
+	if staticCredsRef == nil {
+		return trace.BadParameter("plugin has no credentials reference")
+	}
+
+	req := pluginspb.UpdatePluginStaticCredentialsRequest{
+		Target: &pluginspb.UpdatePluginStaticCredentialsRequest_Query{
+			Query: &pluginspb.CredentialQuery{
+				Labels: staticCredsRef.Labels,
+			},
+		},
+		Credential: &types.PluginStaticCredentialsSpecV1{
+			Credentials: &types.PluginStaticCredentialsSpecV1_APIToken{
+				APIToken: p.updateCreds.awsic.payload,
+			},
+		},
+	}
+
+	_, err = args.plugins.UpdatePluginStaticCredentials(ctx, &req)
```

Comment:
Do we validate credential before it gets to the backend?

Reply:
No, but I'm sure we _could_.

Reply:
...and I had to revert it again because it requires packages in `teleport.e`

Reply:
Hmm, any reason we can't move that to OSS? Validating that anywhere before it gets to the storage backend would do. We have had a couple of support ticket related to failed ic sync due to invalid scim credential. It would be helpful if we can catch such token before its saved in Teleport.

Reply:
Moved `lib/scim/sdk` to OSS.